src/Plugin/RetryPlugin.php

```diff
@@ -35,18 +45,30 @@
      * @param array $config {
      *
      *     @var int $retries Number of retries to attempt if an exception occurs before letting the exception bubble up.
+     *     @var callable $decider A callback that gets a request and an exception to decide if we should retry this or not.
+     *     @var callable $delay A callback to return how many milliseconds we should wait before trying again.
      * }
      */
     public function __construct(array $config = [])
     {
         $resolver = new OptionsResolver();
         $resolver->setDefaults([
             'retries' => 1,
+            'decider' => function (RequestInterface $request, Exception $e) {
+                return true;
```

Comment:
should we check the exception class and only do this when the exception indicates a network problem or a server error? 4xx responses or completely invalid requests don't need to be retried. (though we could want to retry dns errors)

if we make the logic more complicated, we should move this to a static public method of this class instead of the anonymous function, also to allow testing it.